src/client.rs

```diff
@@ -177,169 +174,219 @@ impl Client {
+            return Err(SignatureError::VerificationError {
+                name: "verification error".to_owned(),
+            });
+        }
+    }
+
+    impl Verification for VerifierWithNoError {
+        fn verify(&self, _collection: &Collection) -> Result<(), SignatureError> {
+            Ok(())
+        }
+    }
+
+    impl Verification for VerifierWithInvalidSignatureError {
+        fn verify(&self, _collection: &Collection) -> Result<(), SignatureError> {
+            return Err(SignatureError::InvalidSignature {
+                name: "invalid signature error".to_owned(),
+            });
+        }
+    }
+
+    fn init() {
+        let _ = env_logger::builder().is_test(true).try_init();
+        set_backend(&ReqwestBackend).unwrap();
+    }
+
+    const EXPECTED_VERSION: u64 = 9173;
+
+    fn test_get(
+        mock_server: &MockServer,
+        client: Client,
+        latest_change_response: &str,
+        records_response: &str,
+        expected_result: Result<Vec<serde_json::Value>, ClientError>,
+    ) {
+        let mut get_latest_change_mock = Mock::new()
+            .expect_method(GET)
+            .expect_path("/buckets/monitor/collections/changes/records")
+            .expect_query_param("bucket", "main")
+            .expect_query_param("collection", "url-classifier-skip-urls")
+            .return_status(200)
+            .return_header("Content-Type", "application/json")
+            .return_body(latest_change_response)
+            .create_on(&mock_server);
+
+        let mut get_changeset_mock = Mock::new()
+            .expect_method(GET)
+            .expect_path("/buckets/main/collections/url-classifier-skip-urls/changeset")
+            .expect_query_param("_expected", &EXPECTED_VERSION.to_string())
+            .return_status(200)
+            .return_header("Content-Type", "application/json")
+            .return_body(records_response)
+            .create_on(&mock_server);
+
+        let actual_result = client.get();
+        assert_eq!(1, get_latest_change_mock.times_called());
+        assert_eq!(actual_result, expected_result);
+
+        get_changeset_mock.delete();
+        get_latest_change_mock.delete();
+    }
+
+    #[test]
+    fn test_get_verification() {
+        init();
+
+        let mock_server = MockServer::start();
+
+        let mock_server_address = mock_server.url("");
+
+        let valid_latest_change_response = &format!(
+            "{}{}{}",
+            r#"{
+            "data": [
+                {
+                    "id": "123",
+                    "last_modified": "#,
+            EXPECTED_VERSION,
+            r#",
+                    "bucket":"main",
+                    "collection":"url-classifier-skip-urls",
+                    "host":"localhost:5000"
+                }
+            ]
+        }"#
+        );
+
+        let invalid_latest_change_response = &format!(
+            "{}",
+            r#"{
+            "data": []
+        }"#
+        );
```

Comment:
This is defined too far from its usage. A variable may not be necessary for that